### PR TITLE
Add store conf to have specific ubuntustore

### PR DESCRIPTION
**Pull request by tbille**
# Summary

Fixes https://github.com/CanonicalLtd/snap-squad/issues/588
Add possiblity to store to add a header in the API calls with `X-Ubuntu-Store`.

To do this I had to refactor the code so that blueprints have a constructor: `store_blueprint()` so that we can store parameters (for example `store_query` which is the store to query)

Same thing has been done for the API module which has been transformed into a class so that the header is built once.

# QA

- `./run`
- Should have snapcraft as usual
- In the folder `webapp/configs`, add the file `lime.py`:
```python
WEBAPP_CONFIG = {
    'STORE_QUERY': 'LimeNET',
    'LAYOUT': '_layout-brandstore.html',
    'STORE_NAME': 'LimeNET'
}
```
- ~In `.env` file change `WEBAPP`: `WEBAPP=lime`~
- ~`./run`~
- `./run --env WEBAPP=lime`
- http://0.0.0.0:8004
- Search for lime, you should have a list of snaps
- Search for toto, no snap found

### CODE REVIEW

**Comment by Lukewh** [comment]
Looks good but maybe get @WillMoggridge to take a look too

File: webapp/config.py [hunk 1]
```diff
@@ -24,3 +24,4 @@
 }
 
 WEBAPP = os.getenv('WEBAPP', 'snapcraft')
+WEBAPP_EXTRA_HEADER = {}
```
Comment (WillMoggridge) [inline webapp/config.py]: Can this be made plural please? As the content is not a single item. `WEBAPP_EXTRA_HEADERS`

**Comment by WillMoggridge** [comment]
@tbille You don't need to modify the .env file, that is the beauty of environment variables!

You can run: `./run --env WEBAPP=lime` or `WEBAPP=lime yarn run serve`

**Comment by WillMoggridge** [comment]
I don't think you pass in the required headers anymore, the searches show the normal results.


I think I may have misunderstood the requirement and use of the header. I thought we were talking about misc debug headers like `COMMIT_ID`.
If it is a header that is part of the API specification, it makes sense to set it with an init argument such as:
```python3
api = StoreApi(store='lime')
```

This PR can continue forward. That could be discussed and updated in a follow up PR.

**Pull request merged**